packages/pigment-css-react/README.md

```diff
@@ -854,3 +854,216 @@ function App() {
   )
 }
 ```
+
+## Building a UI library
+
+The key of a UI library is to provide a set of reusable components that can be used and themed across **different projects**.
+
+This guide will show you how to build a statistics component as part of a UI library using Pigment CSS. There are 3 steps to follow:
+
+1. Create the component slots.
+2. Compose slots to create the component.
+3. Style the slot with `ownerState`.
+
+### 1. Create the component slots
+
+Slots let the consumers customize each individual element of the component by targeting its respective name in the [theme's styleOverrides](#themeable-statistics-component).
+
+This statistics component is composed of three slots:
+
+- `root`: the container of the component
+- `value`: the number of the statistics
+- `unit`: the unit or description of the statistics
+
+> 💡 Though you can give these slots any names you prefer, we recommend using `root` for the outermost container element for consistency with the rest of the library.
+
+Use the `styled` API with `name` and `slot` parameters to create the slots, as shown below:
+
+```js
+import * as React from 'react';
+import { styled } from '@pigment-css/react';
+
+const StatRoot = styled('div', {
+  name: 'MuiStat', // The component name
+  slot: 'root', // The slot name
+})(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '1rem',
+  padding: '0.75rem 1rem',
+  backgroundColor: theme.colors.primary.background,
+  borderRadius: theme.radius.xs,
+  boxShadow: theme.shadow.sm,
+  letterSpacing: '-0.025em',
+  fontWeight: 600,
+}));
+
+const StatValue = styled('div', {
+  name: 'MuiStat',
+  slot: 'value',
+})(({ theme }) => ({
+  ...theme.typography.h3,
+}));
+
+const StatUnit = styled('div', {
+  name: 'MuiStat',
+  slot: 'unit',
+})(({ theme }) => ({
+  ...theme.typography.body2,
+  color: theme.colors.neutral.foreground,
+}));
+```
+
+### 2. Create the component
+
+Assemble the component using the slots created in the previous step:
+
+```js
+// /path/to/Stat.js
+import * as React from 'react';
+
+const StatRoot = styled('div', {
+  name: 'MuiStat',
+  slot: 'root',
+})(…);
+
+const StatValue = styled('div', {
+  name: 'MuiStat',
+  slot: 'value',
+})(…);
+
+const StatUnit = styled('div', {
+  name: 'MuiStat',
+  slot: 'unit',
+})(…);
```

Comment:
These are redundant here given they are defined just above.

Reply:
Yeah, was just about to ask what's the different between this section and previous, too